addons/web/static/tests/views/fields/many2many_tags_field_tests.js

```diff
@@ -1947,4 +1947,52 @@ QUnit.module("Fields", (hooks) => {
         assert.verifySteps(["name search with context given", "read with context given"]);
         assert.strictEqual(target.querySelector(".o_field_tags").innerText, "gold coucou");
     });
+
+    QUnit.test("Many2ManyTagsField don't use virtualId for 'name_search'", async (assert) => {
+        await makeView({
+            type: "form",
+            resModel: "partner",
+            serverData,
+            arch: `<form>
+                <field name="turtles" widget="many2many_tags"/>
+                <field name="turtles">
+                    <tree>
+                        <field name="display_name"/>
+                    </tree>
+                    <form>
+                        <field name="display_name"/>
+                    </form>
+                </field>
+            </form>`,
+            async mockRPC(route, { method, kwargs }) {
+                if (method === "name_search") {
+                    assert.step("name_search");
+                    // no virtualId in domain
+                    assert.deepEqual(kwargs.args, ["!", ["id", "in", []]]);
```

Comment:
maybe add a record in the relation in the data before spawning the view, to show that real ids are indeed there